Velaptor/Graphics/RenderMediator.cs

```diff
@@ -4,41 +4,224 @@
 
 namespace Velaptor.Graphics;
 
+using System;
+using System.Collections.Generic;
+using Batching;
 using Carbonate.NonDirectional;
 using Factories;
 using Guards;
+using OpenGL.Batching;
 
 /// <inheritdoc/>
 internal sealed class RenderMediator : IRenderMediator
 {
     private readonly IPushReactable pushReactable;
+    private readonly IBatchPullReactable<TextureBatchItem> texturePullReactable;
+    private readonly IBatchPullReactable<FontGlyphBatchItem> fontPullReactable;
+    private readonly IBatchPullReactable<RectBatchItem> rectPullReactable;
+    private readonly IBatchPullReactable<LineBatchItem> linePullReactable;
+    private readonly IRenderBatchReactable<TextureBatchItem> textureRenderBatchReactable;
+    private readonly IRenderBatchReactable<FontGlyphBatchItem> fontRenderBatchReactable;
+    private readonly IRenderBatchReactable<RectBatchItem> rectRenderBatchReactable;
+    private readonly IRenderBatchReactable<LineBatchItem> lineRenderBatchReactable;
+    private readonly IComparer<RenderItem<TextureBatchItem>> textureItemComparer;
+    private readonly IComparer<RenderItem<FontGlyphBatchItem>> fontItemComparer;
+    private readonly IComparer<RenderItem<RectBatchItem>> rectItemComparer;
+    private readonly IComparer<RenderItem<LineBatchItem>> lineItemComparer;
+    private readonly IDisposable endBatchUnsubscriber;
+    private readonly IDisposable shutDownUnsubscriber;
+
+    // The total amount of layers supported
+    private readonly Memory<int> allLayers = new (new int[1000]);
 
     /// <summary>
     /// Initializes a new instance of the <see cref="RenderMediator"/> class.
     /// </summary>
     /// <param name="reactableFactory">Creates reactables for sending and receiving notifications with or without data.</param>
-    public RenderMediator(IReactableFactory reactableFactory)
+    /// <param name="textureItemComparer">Compares 2 texture batch items for the purpose of sorting.</param>
+    /// <param name="fontItemComparer">Compares 2 font batch items for the purpose of sorting.</param>
+    /// <param name="rectItemComparer">Compares 2 rect batch items for the purpose of sorting.</param>
+    /// <param name="lineItemComparer">Compares 2 line batch items for the purpose of sorting.</param>
```

Comment:
Change to two, not "2".
```suggestion
    /// <param name="textureItemComparer">Compares two texture batch items for the purpose of sorting.</param>
    /// <param name="fontItemComparer">Compares two font batch items for the purpose of sorting.</param>
    /// <param name="rectItemComparer">Compares two rect batch items for the purpose of sorting.</param>
    /// <param name="lineItemComparer">Compares two line batch items for the purpose of sorting.</param>
```